lib/App.js

```diff
@@ -51,7 +61,7 @@ class App {
       route = new Route(route);
     }
 
-    this._middleware.push(route);
+    this._middleware.push({ route, errorMiddleware : false });
```

Comment:
Rather than pushing a bool of `errorMiddleware`, let's just push `{method : 'then' | 'catch' }`